src/App.tsx

```diff
@@ -1,27 +1,60 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goodsList, setGoodsList] = useState<Good[]>([]);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const loadAllGoods = async () => {
+    const allGoods = await getAll();
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+    setGoodsList(allGoods);
+  };
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+  const load5Goods = async () => {
+    const first5Goods = await get5First();
 
-    <GoodsList goods={[]} />
-  </div>
-);
+    setGoodsList(first5Goods);
+  };
+
+  const loadRedGoods = async () => {
+    const redGoods = await getRedGoods();
+
+    setGoodsList(redGoods);
+  };
```

Comment:
Nice job. What about creating one function that takes the function, calls it, and adds data to the state?